### PR TITLE
Receiver component

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: common/network/iota_packet.h [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
+#include <stdlib.h>
+
+#include "common/network/endpoint.h"
 
 #define TRANSACTION_PACKET_SIZE 1650
 
 typedef struct {
-  char data[TRANSACTION_PACKET_SIZE];
-  socket_addr_t source;
+  char content[TRANSACTION_PACKET_SIZE + 1];
```
Comment (th0br0) [inline common/network/iota_packet.h:18]: why the `+1`? it's a fixed size packet / you already have the `length` field in this struct?

File: common/network/components/receiver.h [hunk 2]
```diff
@@ -10,18 +10,25 @@
...
 
+#include "common/network/iota_packet.h"
 #include "common/thread_handle.h"
 
 typedef struct {
   thread_handle_t thread;
   bool running;
+  void *opaque_network;
+  uint16_t tcp_port;
+  uint16_t udp_port;
```
Comment (th0br0) [inline common/network/components/receiver.h]: what are they needed for?

File: common/network/services/receiver.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (th0br0) [inline common/network/services/receiver.cpp]: could we suffix C++ files with `.cc`?

File: common/network/services/receiver.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+bool receiver_service_start(receiver_state_t* const state) {
+  if (state == NULL) {
+    return false;
+  }
+  try {
+    boost::asio::io_context ctx;
+    state->opaque_network = &ctx;
+    TcpReceiverService tcpService(state, ctx, state->tcp_port);
```
Comment (th0br0) [inline common/network/services/receiver.cpp]: maybe disable the individual service if port is `== 0`?

File: common/network/services/receiver.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  packet->source.protocol = protocol;
+  packet->content[length] = '\0';
+}
+
+bool receiver_service_start(receiver_state_t* const state) {
+  if (state == NULL) {
+    return false;
+  }
+  try {
+    boost::asio::io_context ctx;
```
Comment (th0br0) [inline common/network/services/receiver.cpp]: who owns the thread running the context's event loop?
shouldn't the two receivers run on separate threads?

Comment (th0br0) [inline common/network/services/receiver.cpp]: who owns the thread running the context's event loop?
shouldn't the two receivers run on separate threads?

File: common/network/services/udp_receiver.hpp [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+#pragma once
+
+#include <boost/asio.hpp>
+
+#include "common/network/components/receiver.h"
+
+class UdpReceiverService {
+ public:
+  UdpReceiverService(receiver_state_t* const state,
```
Comment (th0br0) [inline common/network/services/udp_receiver.hpp]: `clang-format`?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/network/services/udp_receiver.hpp (hunk 6 above)
Comment (thibault-martinez) [reply common/network/services/udp_receiver.hpp]: It is not triggered, what were you referring to ?

File: common/network/iota_packet.h (hunk 1 above)
Comment (thibault-martinez) [reply common/network/iota_packet.h:18]: To hold a '\0' for printing purpose

**Comment by thibault-martinez** [comment]
I changed a lot of things here.
Receiver services are now more independent and communicate by pushing packets to a queue.
I hooked the UDP and TCP queues from the receiver component to the processor component queue.

File: common/network/queues/concurrent_queue_packet.h [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __COMMON_NETWORK_CONCURRENT_QUEUE_PACKET_H__
+#define __COMMON_NETWORK_CONCURRENT_QUEUE_PACKET_H__
+
+#include "common/network/iota_packet.h"
+#include "concurrent_queue.h.inc"
```
Comment (thibault-martinez) [inline common/network/queues/concurrent_queue_packet.h]: Absolute path

File: ciri/node.c [hunk 8]
```diff
@@ -8,6 +8,10 @@
 #include "ciri/node.h"
 #include "common/network/logger.h"
 
+// TODO(thibault) configuration variable
+static uint16_t tcp_port_g = 4242;
```
Comment (thibault-martinez) [inline ciri/node.c]: 14265

File: ciri/node.c [hunk 9]
```diff
@@ -8,6 +8,10 @@
...
+static uint16_t udp_port_g = 4243;
```
Comment (thibault-martinez) [inline ciri/node.c]: 14265

File: ciri/node.c [hunk 10]
```diff
@@ -8,6 +8,10 @@
 #include "ciri/node.h"
 #include "common/network/logger.h"
 
+// TODO(thibault) configuration variable
```
Comment (th0br0) [inline ciri/node.c:11]: could you file an issue in addition to track this?

File: common/network/iota_packet.h (hunk 1 above)
Comment (th0br0) [reply common/network/iota_packet.h:18]: where do you print it though?

**Review by th0br0 — APPROVED** [review]

File: common/network/iota_packet.h (hunk 1 above)
Comment (thibault-martinez) [reply common/network/iota_packet.h:18]: Locally when packets are received. I think it'll be useful for some time, we'll defo be able to remove it when the whole network part is done.

**Pull request merged**